build_tools/buildkite/pipelines/partial/update-pipelines.yml

```diff
@@ -0,0 +1,35 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+agents:
+  queue: "orchestration"
+  security: "submitted"
+
+steps:
+  # TODO genericize this
+  - group: "Updating pipelines"
+    # Each of these are different steps so they can succeed and fail separately,
+    # which means that step success status indicates whether the pipeline was
+    # successfully updated. That ties into how the update script checks for
+    # updates from later builds.
+    steps:
```

Comment:
Do the following two steps run in the same transaction or two transactions?

According to https://buildkite.com/docs/pipelines/controlling-concurrency, it looks like we don't prevent another pipeline update from other builds happens between the presubmit update and postsubmit update.

This might be fine, just want to double check : )

Reply:
Separate transactions, and that's deliberate. We can't actually make multiple updates a transaction together because we don't have actual transaction APIs with Buildkite. If we did it that way, one of the pipeline updates could fail and then we would be in a partial state where the check if a later build had already completed wouldn't be possible.